### PR TITLE
Hotfix azurermwebapp next link

### PR DESCRIPTION
**Pull request by Ajay-MS**

### CODE REVIEW

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts [hunk 1]
```diff
@@ -129,25 +131,30 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
     return deferred.promise;
 }
 
-async function getAzureRMWebAppID(SPN, webAppName: string, resourceType: string) {
+async function getAzureRMWebAppID(SPN, webAppName: string,url: string) {
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: space after comma

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts [hunk 2]
```diff
@@ -129,25 +131,30 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
...
             var webAppIDDetails = JSON.parse(body);
-            deferred.resolve(webAppIDDetails.value[0]);
+            if((webAppIDDetails.value.length == 0) && (webAppIDDetails.nextLink)){
+                tl.debug("Requesting nextLink to accesss webappId for webapp " + webAppName);
+                deferred.resolve(await getAzureRMWebAppID(SPN, webAppName, webAppIDDetails.nextLink));
+            } else {
+                if(webAppIDDetails.value.length == 0){
+                    deferred.reject(tl.loc("WebAppDoesntExist", webAppName));
+                }
+                deferred.resolve(webAppIDDetails.value[0]);
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: change it to elseif and else

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts [hunk 3]
```diff
@@ -129,25 +131,30 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
     return deferred.promise;
 }
 
-async function getAzureRMWebAppID(SPN, webAppName: string, resourceType: string) {
+async function getAzureRMWebAppID(SPN, webAppName: string,url: string) {
 
     var deferred = Q.defer<any>();
     var accessToken = await getAuthorizationToken(SPN);
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: Access token shouldn't be fetched for each call, do this outside. Verify access token expiration time as well.

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts [hunk 4]
```diff
@@ -129,25 +131,30 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
...
+    httpObj.get('GET', url, headers, async (error, response, body) => {
         if(error) {
             deferred.reject(error);
         }
         else if(response.statusCode === 200) {
             var webAppIDDetails = JSON.parse(body);
-            deferred.resolve(webAppIDDetails.value[0]);
+            if((webAppIDDetails.value.length == 0) && (webAppIDDetails.nextLink)){
+                tl.debug("Requesting nextLink to accesss webappId for webapp " + webAppName);
+                deferred.resolve(await getAzureRMWebAppID(SPN, webAppName, webAppIDDetails.nextLink));
```
Comment (mvvsubbu) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: This one I believe we should try to initiate a conversation with azure team on why nextlink is being sent and we are depending our recursive condition to exit on successful return of the value. We should validate that assume will not cause any issues

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts (hunk 3 above)
Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: Access Token expires after one hour. 
  Access token Response  has `expiresIn: 3600` (sec)  property.

File: Tasks/AzureRmWebAppDeployment/azurermutil.ts [hunk 5]
```diff
@@ -129,32 +135,31 @@ function getAuthorizationToken(SPN): Q.Promise<string> {
     return deferred.promise;
 }
 
-async function getAzureRMWebAppID(SPN, webAppName: string, resourceType: string) {
-
+async function getAzureRMWebAppID(SPN, webAppName: string, url: string, headers) {
     var deferred = Q.defer<any>();
     var accessToken = await getAuthorizationToken(SPN);
```
Comment (kmkumaran) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: Is this get token call still required?

Comment (vincentdass) [inline Tasks/AzureRmWebAppDeployment/azurermutil.ts]: removed.

**Pull request merged**